design-documents/simplified-composer-project.md

```diff
@@ -0,0 +1,172 @@
+# Simplified Composer Project
+
+Composer project is intended for production use.
+Development needs can be covered by a custom/adopted Composer project.
+We can provide such `composer.json` in documentation.
+
+## Proposed Changes on High-Level
+
+1. Remove `dev` sections
+1. Remove `autoload` parts that cover development scenarios, such as git-based Magento paths
+1. Move component-specific parts to corresponding Composer packages. See examples in "Changes" section
+
+As a result the project becomes simple and would have fewer reasons to change ([Single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle).
+Responsibilities move to packages really containing corresponding code instead of delegating it to the project's `composer.json`.
+
+## Deployment scenarios
+
+All scenarios are focused on production deployment and so should lead to absence of development resources (wherever possible).
+
+### Initial installation
+
+Performed as described on [DevDocs](https://devdocs.magento.com/guides/v2.3/install-gde/composer.html):
+```
+composer create-project --repository=https://repo.magento.com/ magento/project-community-edition <install-directory-name>
+```
+
+### Upgrade with the new project structure
+
+```
+composer require magento/product-community-edition <new-product-version>
+```
+
+### Upgrade with the old project structure and preserve it
+
+Discouraged.
+Update all the dependencies manually.
+
+### Upgrade from the old project structure to the new one
+
+Modify `composer.json` to eliminate non-production dependencies, as described in the proposal or use plugin that will do it automatically.
+After that use approach described in "Upgrade with the proposed project structure".
+
+## Low-level changes
+
+Based on current `composer.json` the following changes are proposed:
+1. Remove `require-dev` section
+1. Move `"conflict": { "gene/bluefoot": "*" }` to `magento/page-builder` module
+1. Remove `"Magento\\Framework\\": "lib/internal/Magento/Framework/"`, `"Magento\\": "app/code/Magento/"` and `"app/code/"` from `autoload` section
```

Comment:
Can you clarify how it will affect development experience for developers who develop its own extension in app/code?